doc/block_roadmap.md

```diff
@@ -0,0 +1,77 @@
+## Overview
+
+We use `Block` to describe our neural network's topology. Actually, a neural network will be executed in various hardware environment. Generally, we will implement data parallelism and model parallelism to get high performance and scalability.
```

Comment:
We might need to be clear about the problem setting here.

> We use `Block` to describe our neural network's topology. 

==>

> PaddlePaddle represents a neural network as a *block*.  A block is like a pair of curly braces in programming languages like C++ and Java, it includes some variables and a sequence of instructions, or, operators. 
>
> Usually we run a block using a CPU thread. But we want to accelerate the execution of some operators using devices like GPU and FPGA.

Reply:
Done